### ISSUE TITLE
[KV Feedback]: how to handle date/time

### ISSUE THREAD

**Issue by iuioiua** [denoland/deploy_feedback#416]
### 🔍

- [X] Did you search for existing issues?

### Type of feedback

Other

### Description

Since KV is a distributed database, what is the recommended method of dealing with times? I'd assume one must always set/get in formats that "respect" (if that's the right word) UTC time.
```ts
const post = {
  id: crypto.randomUUID(),
  createdAt: new Date(),
}

const postsKey = ["posts", post.id];
const postsByTime = ["posts_by_time", post.createdAt.getTime()]

await kv.atomic()
  .set(postsKey, post)
  .set(postsByTime, post)
```

**Comment by shemarmc** [comment]
I believe the following code from Deno SaaSKit has the answers you're looking for.
```javascript
export async function createItem(item: Item) {
  const itemsKey = ["items", item.id];
  const itemsByTimeKey = ["items_by_time", item.createdAt.getTime(), item.id];
  const itemsByUserKey = ["items_by_user", item.userId, item.id];
  const itemsCountKey = ["items_count", formatDate(new Date())];

  // ...
```
https://github.com/denoland/saaskit/blob/520a4048265af18f08118a7bb83516d449bfa92b/utils/db.ts#L103

```javascript
/** Converts `Date` to ISO format that is zero UTC offset */
export function formatDate(date: Date) {
  return date.toISOString().split("T")[0];
}
```

Here `getTime()` is used as well as a formatted string representing the date in UTC. My recommendation is yes use UTC format for date in your database. However, a date-time string can always be converted when stored in full.

Edit: In addition "The `getTime()` method of [Date](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) instances returns the number of milliseconds for this date since the [epoch](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#the_epoch_timestamps_and_invalid_date), which is defined as the midnight at the beginning of January 1, 1970, UTC."
In other words `getTime()` will always present a time respecting UTC time.

**Comment by iuioiua** [comment]
Sorry, I didn't clarify that I was already aware of `Date.getTime()`. I was more wondering how `post` is stored if the `createdAt` property is of type `Date`.
